src/main/java/com/breskul/bibernate/persistence/Persistence.java

```diff
@@ -1,17 +1,52 @@
 package com.breskul.bibernate.persistence;
 
-import org.postgresql.ds.PGSimpleDataSource;
+import com.breskul.bibernate.config.PropertiesConfiguration;
+import com.breskul.bibernate.exception.BibernateException;
+import java.util.Objects;
 
-// TODO: javadoc
+/**
+ * The Persistence class provides a convenient way to create a Bibernate SessionFactory based on configuration properties. It retrieves
+ * connection details from a properties file using {@link PropertiesConfiguration} and initializes a {@link BibernateDataSource}
+ * accordingly. The default behavior is to use a generic DataSource, but you can specify a custom JDBC driver class if needed.
+ * <p>
+ * Usage Example:
+ * <pre>
+ *   {@code
+ *  SessionFactory sessionFactory = Persistence.createSessionFactory();
+ * }
+ * </pre>
+ */
 public class Persistence {
 
-  // TODO: read properties for data source
+  private static final String CONNECTION_URL = "bibernate.connection.url";
+  private static final String CONNECTION_USERNAME = "bibernate.connection.username";
+  private static final String CONNECTION_PASSWORD = "bibernate.connection.password";
+  private static final String CONNECTION_DRIVER_CLASS = "bibernate.connection.driver_class";
```

Comment:
I suggest not to use the underscore and name property jusr as "bibernate.connection.driver''